js/src/forum/components/Search.tsx

```diff
@@ -76,7 +76,7 @@ export default class Search<T extends SearchAttrs = SearchAttrs> extends Compone
     return this.searchState;
   }
   protected set state(state: SearchState) {
-    fireDeprecationWarning('`state` property of the Search component is deprecated', '3212');
+    state !== undefined && fireDeprecationWarning('`state` property of the Search component is deprecated', '3212');
```

Comment:
```suggestion
    // Workaround to prevent triggering deprecation warnings due to Mithril
    // setting state to undefined when creating components
    state !== undefined && fireDeprecationWarning('`state` property of the Search component is deprecated', '3212');
```